### PR TITLE
MSLE rather than MSE in the loss function

### PR DESCRIPTION
**Pull request by goldmermaid**
Fixes of the loss function in both MXNet and PyTorch version: changing MSE to MSLE (mean squared log error). @AnirudhDagar can you help review this PR?

### CODE REVIEW

**Comment by astonzhang** [comment]
This is a very simple exercise so we intend not to provide an ideal solution in the demo version. Thus we want to keep it simple and encourage readers to improve the demo in the exercises.

**Comment by mli** [comment]
Job d2l-en/PR-1144/2 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1144/

**Comment by mli** [comment]
Job d2l-en/PR-1144/3 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1144/

**Comment by mli** [comment]
Job d2l-en/PR-1144/4 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1144/

**Comment by goldmermaid** [comment]
> This is a very simple exercise so we intend not to provide an ideal solution in the demo version. Thus we want to keep it simple and encourage readers to improve the demo in the exercises.

Then shall we explain to the readers why we use L2 for optimization and use MSLE for calculating the loss? Besides, we may need to modify the TF version back, so that the 3 versions are aligned.

> Besides, please test the code locally when there's any change of code before sending a PR :)

Sorry, will test all frameworks next time! :)

**Comment by astonzhang** [comment]
@terrytangyuan Can you change the loss function (to be optimized) from log rmse to squared loss? This will be consistent with the other implementations.

**Comment by terrytangyuan** [comment]
I am a little confused. i see that this PR changes all loss to be rmse and TF is already using `log_rmse`. Could you clarify?

**Comment by goldmermaid** [comment]
> I am a little confused. i see that this PR changes all loss to be rmse and TF is already using `log_rmse`. Could you clarify?

Hey @terrytangyuan , no worries. I just changed it. Please review the modifications.

**Comment by astonzhang** [comment]
LGTM. Thanks Rachel! @terrytangyuan PTAL

**Comment by mli** [comment]
Job d2l-en/PR-1144/6 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1144/

**Comment by AnirudhDagar** [comment]
@goldmermaid looks like an issue with the loss plots. See [section 4.10.7](http://preview.d2l.ai/d2l-en/PR-1144/chapter_multilayer-perceptrons/kaggle-house-price.html#model-selection)

**Comment by goldmermaid** [comment]
Good catch @AnirudhDagar . Fixed it!

**Comment by mli** [comment]
Job d2l-en/PR-1144/8 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1144/

**Comment by mli** [comment]
Job d2l-en/PR-1144/9 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1144/

**Comment by terrytangyuan** [comment]
@goldmermaid Seems like the loss is too low now 0.02 (actual) vs around 0.16 (expected).

**Comment by mli** [comment]
Job d2l-en/PR-1144/10 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1144/

**Comment by mli** [comment]
Job d2l-en/PR-1144/11 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1144/

**Comment by goldmermaid** [comment]
@terrytangyuan Fixed!

**Comment by terrytangyuan** [comment]
Thanks!!
